/*
 * This file was automatically generated by EvoSuite
 * Thu May 29 01:07:36 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution1478;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Solution1478_ESTest extends Solution1478_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[8];
      intArray0[1] = (-4048);
      // Undeclared exception!
      try { 
        solution1478_0.minDistance(intArray0, 0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[8];
      intArray0[2] = 343;
      // Undeclared exception!
      try { 
        solution1478_0.minDistance(intArray0, 0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[34];
      // Undeclared exception!
      solution1478_0.minDistance(intArray0, 1364);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      // Undeclared exception!
      try { 
        solution1478_0.minDistance((int[]) null, (-4377));
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[1];
      // Undeclared exception!
      try { 
        solution1478_0.minDistance(intArray0, (-5590));
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // -5589
         //
         verifyException("original.Solution1478", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[7];
      int int0 = solution1478_0.minDistance(intArray0, 1);
      assertEquals(0, int0);
      assertArrayEquals(new int[] {0, 0, 0, 0, 0, 0, 0}, intArray0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      Solution1478 solution1478_0 = new Solution1478();
      int[] intArray0 = new int[4];
      int int0 = solution1478_0.minDistance(intArray0, 1985);
      assertEquals(1073741824, int0);
  }
}
